cli/endpoints/batch/deploy-pipelines/batch-scoring-with-preprocessing/components/prepare/src/prepare.py

```diff
@@ -0,0 +1,188 @@
+) -> ColumnTransformer:
+    # Configure the categorical encoder
+    if categorical_encoding == "ordinal":
+        categorical_encoder = OrdinalEncoder(
+            handle_unknown="use_encoded_value", unknown_value=np.nan
+        )
+    elif categorical_encoding == "onehot":
+        categorical_encoder = OneHotEncoder(handle_unknown="ignore")
+    else:
+        raise ValueError(
+            f"categorical_encoding '{categorical_encoding}' is not a valid encoding strategy. Possible values are 'ordinal' or 'onehot'"
+        )
+
+    # For continuous variables, replace missing values with the median and then normalize by
+    # subtracting the mean and dividing by the standard deviation
+    continuous_pipeline = sklearn.pipeline.Pipeline(
+        [("imputer", SimpleImputer(strategy="median")), ("scaler", StandardScaler())]
+    )
+
+    # For discrete variables, encode the data
+    discrete_pipeline = sklearn.pipeline.Pipeline(
+        [
+            ("imputer", SimpleImputer(strategy="most_frequent")),
+            ("encoder", categorical_encoder),
+        ]
+    )
+
+    # Build the pipeline
+    transformations = ColumnTransformer(
+        [
+            ("continuous_pipe", continuous_pipeline, continuous_features),
+            ("discrete_pipe", discrete_pipeline, discrete_features),
+        ],
+        remainder="passthrough",
+    )  # Target will passthrough if there
+
+    return transformations
+
+
+def preprocess_heart_disease_data(
+    df,
+    continuous_features,
+    discrete_features,
+    target,
+    categorical_encoding: str = "ordinal",
+    transformations=None,
+):
+    mlflow.sklearn.autolog()
+
+    if target in df.columns:
+        features_df = df.drop(columns=[target])
+        restore_target = True
+    else:
+        features_df = df
+        restore_target = False
+
+    if transformations:
+        df_transformed = transformations.transform(features_df)
+    else:
+        transformations = build_preprocessing_pipeline(
+            categorical_encoding=categorical_encoding,
+            continuous_features=continuous_features,
+            discrete_features=discrete_features,
+        )
+        df_transformed = transformations.fit_transform(features_df)
+
+    # Get columns names from transformations
+    transformed_discrete_features = (
+        transformations.transformers_[1][1]
+        .named_steps["encoder"]
+        .get_feature_names_out(discrete_features)
+    )
+    all_features = continuous_features + list(transformed_discrete_features)
+
+    if restore_target:
+        target_values = df[target].to_numpy().reshape(len(df), 1)
+        df_transformed = np.hstack((df_transformed, target_values))
+        all_features.append(target)
+
+    return pd.DataFrame(df_transformed, columns=all_features), transformations
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("prepare")
+    parser.add_argument("--data_path", type=str, help="Path to the data to transform")
```

Comment:
follow the same style convention for the arg parsers - I'd recomment running the code through "black"

`pip install black`
`black . `

Reply:
I used black!